extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -1,122 +1,110 @@
-    from cyclonedx.model import ExternalReference, ExternalReferenceType, XsUri
-    name_ = name(n)
-    purl = package_url(n, name_)
-    result = Component(
-        type=package_type_to_component_type(n.conanfile.package_type),
-        name=name_,
-        version=n.conanfile.version,
-        licenses=licenses(n.conanfile.license),
-        bom_ref=purl.to_string(),
-        purl=purl,
-        description=n.conanfile.description
-    )
-    if n.conanfile.homepage:
-        result.external_references.add(ExternalReference(
-            type=ExternalReferenceType.WEBSITE,
-            url=XsUri(n.conanfile.homepage),
-        ))
-    return result
+    from cyclonedx.model.bom import Bom
 
 
-def format_cyclonedx_14_json(result):
+def format_cyclonedx_14_json(bom: 'Bom') -> None:
     from cyclonedx.output.json import JsonV1Dot4
-    serialized_json = JsonV1Dot4(result).output_as_string()
+    serialized_json = JsonV1Dot4(bom).output_as_string()
     cli_out_write(serialized_json)
 
 
-def format_text(result):
-    out = ConanOutput()
-    out.error("Format 'text' not supported")
-
-
-def me_as_tool():
-    from cyclonedx.model import ExternalReference, ExternalReferenceType, XsUri
-    from cyclonedx.model.bom import Tool
-    result = Tool(name="conan extension recipe:create-sbom")
-    result.external_references.add(ExternalReference(
-        type=ExternalReferenceType.WEBSITE,
-        url=XsUri("https://github.com/conan-io/conan-extensions")))
-    return result
+def format_text(_: Any) -> None:
+    raise ConanException("Format 'text' not supported")
 
 
 @conan_command(group="Recipe", formatters={
     "text": format_text,  # added by default in BaseConanCommand.__init__
     "cyclonedx_1.4_json": format_cyclonedx_14_json})
-def create_sbom(conan_api: ConanAPI, parser, *args):
-    """
-    creates an SBOM in CycloneDX 1.4 JSON format
-    """
+def create_sbom(conan_api: ConanAPI, parser, *args) -> 'Bom':
+    """Create a CycloneDX Software Bill of Materials (SBOM)"""
+
     try:
-        import cyclonedx
+        from cyclonedx.factory.license import LicenseFactory
+        from cyclonedx.model import (ExternalReference, ExternalReferenceType,
+                                     LicenseChoice, Tool, XsUri)
+        from cyclonedx.model.bom import Bom
+        from cyclonedx.model.component import Component, ComponentType
+        from packageurl import PackageURL
     except ModuleNotFoundError:
-        sys.stderr.write(
-            "The sbom extension needs an additional package, please run 'pip install cyclonedx-python-lib'\n")
+        # Assert on RUNTIME of the actual conan-command, that all requirements exist.
+        # Since conan loads all extensions when started, this check could prevent conan from running,
+        # if loading dependencies is performed outside the actual conan-command in global/module scope.
+        print('The sbom extension needs an additional package, please run:'
```

Comment:
@hedtke I forgot a comma(`,`) at the end of this line, right after `please run:'`
Could you add it for me? 

my idea was, that every paras of `print()` would be a new line, ths is why the `, sep='\n'` is at the end

Reply:
THanks.

this was done via ee19212cfecf295888c83eea8c6a36cdda81a7d8